### PR TITLE
show-error-message-when-server-is-unresponsive

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
image
image

### CODE REVIEW

File: app/scripts/main.js [hunk 1]
```diff
@@ -249,6 +249,10 @@ mainModule.controller('mainController', function ($window, $http, $scope, $rootS
                 })
                 .catch(function (response) {
                     console.error('Error checking if session is valid:', response);
+                    var connectionError = $('<div id="connection-error"><label style="color:red; font-weight:bold; font-size:15px; padding-left:30px;">The server cannot be reached. Check first your network connection, and potentially the server status.</label></div>');
```
Comment (medou-boushab) [inline app/scripts/main.js:252]: better to use a directive ng-if or ng-show

Comment (medou-boushab) [reply app/scripts/main.js:252]: please use what we have in inspian to be consistent with that used in submission view alerts.

**Review by mboussaa — APPROVED** [review]

**Review by amouhoub — CHANGES_REQUESTED** [review]
Yes, as @medou-boushab pointed out, angular hates manipulating elements with jQuery.
Use a scope variable for when you get errors and display the message directly in HTML using ng-if

**Comment by oanaBiancaSchiopu** [comment]
> Yes, as @medou-boushab pointed out, angular hates manipulating elements with jQuery. Use a scope variable for when you get errors and display the message directly in HTML using ng-if

@amouhoub is your comment still up to date? I have commited some changes after the comment of Medou

**Review by amouhoub — APPROVED** [review]

**Pull request merged**